sdks/python/apache_beam/io/filesystem.py

```diff
@@ -0,0 +1,367 @@
+#
+# Licensed to the Apache Software Foundation (ASF) under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+"""File system abstraction for file based sources and sinks."""
+
+from __future__ import absolute_import
+
+import bz2
+import cStringIO
+import os
+import zlib
+
+
+DEFAULT_READ_BUFFER_SIZE = 16 * 1024 * 1024
+
+
+class _CompressionType(object):
```

Comment:
Either use Python's enum module, or just use Strings here. This manual layer of wrapping isn't providing much value.